File: problems/java-palindrome-checker/src/test/java/PalindromeCheckerTest.java
import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;

class PalindromeCheckerTest {
    @Test void testClassicPalindrome() {
        assertTrue(PalindromeChecker.isPalindrome("A man a plan a canal Panama"));
    }

    @Test void testSimplePalindrome() {
        assertTrue(PalindromeChecker.isPalindrome("racecar"));
    }

    @Test void testNotPalindrome() {
        assertFalse(PalindromeChecker.isPalindrome("hello"));
    }

    @Test void testEmptyString() {
        assertTrue(PalindromeChecker.isPalindrome(""));
    }

    @Test void testSingleChar() {
        assertTrue(PalindromeChecker.isPalindrome("a"));
    }

    @Test void testNull() {
        assertFalse(PalindromeChecker.isPalindrome(null));
    }

    @Test void testNumericPalindrome() {
        assertTrue(PalindromeChecker.isPalindrome("12321"));
    }

    @Test void testWithPunctuation() {
        assertTrue(PalindromeChecker.isPalindrome("Was it a car or a cat I saw?"));
    }
}

File: problems/dotnet-palindrome-checker/PalindromeCheckerTest.cs
using System;
using Xunit;

public class PalindromeCheckerTest
{
    [Fact]
    public void TestBasicPalindrome()
    {
        Assert.True(PalindromeChecker.IsPalindrome("racecar"));
    }

    [Fact]
    public void TestNotPalindrome()
    {
        Assert.False(PalindromeChecker.IsPalindrome("hello"));
    }

    [Fact]
    public void TestWithSpaces()
    {
        Assert.True(PalindromeChecker.IsPalindrome("A man a plan a canal Panama"));
    }

    [Fact]
    public void TestWithPunctuation()
    {
        Assert.True(PalindromeChecker.IsPalindrome("A man, a plan, a canal: Panama"));
    }

    [Fact]
    public void TestCaseInsensitive()
    {
        Assert.True(PalindromeChecker.IsPalindrome("Racecar"));
    }

    [Fact]
    public void TestEmptyString()
    {
        Assert.True(PalindromeChecker.IsPalindrome(""));
    }

    [Fact]
    public void TestNullString()
    {
        Assert.True(PalindromeChecker.IsPalindrome(null));
    }

    [Fact]
    public void TestSingleCharacter()
    {
        Assert.True(PalindromeChecker.IsPalindrome("a"));
    }

    [Fact]
    public void TestTwoCharacters()
    {
        Assert.True(PalindromeChecker.IsPalindrome("aa"));
        Assert.False(PalindromeChecker.IsPalindrome("ab"));
    }

    [Fact]
    public void TestNumbers()
    {
        Assert.True(PalindromeChecker.IsPalindrome("12321"));
        Assert.False(PalindromeChecker.IsPalindrome("12345"));
    }
}
